chromadb/api/types.py

```diff
@@ -515,39 +524,46 @@ def validate_embeddings(embeddings: Embeddings) -> Embeddings:
     return embeddings
 
 
-def validate_batch(
-    batch: Tuple[
-        IDs,
-        Optional[Embeddings],
-        Optional[Metadatas],
-        Optional[Documents],
-        Optional[URIs],
-    ],
+def validate_batch_size(
+    record_set: RecordSet,
     limits: Dict[str, Any],
 ) -> None:
-    if len(batch[0]) > limits["max_batch_size"]:
+    (_, batch_size) = get_n_items_from_record_set(record_set)
+
+    if batch_size > limits["max_batch_size"]:
         raise ValueError(
-            f"Batch size {len(batch[0])} exceeds maximum batch size {limits['max_batch_size']}"
+            f"Batch size {batch_size} exceeds maximum batch size {limits['max_batch_size']}"
         )
 
 
+def get_n_items_from_record_set(record_set: RecordSet) -> Tuple[str, int]:
+    """
+    Get the number of items in the record set.
+    """
+    for field, value in record_set.items():
+        if isinstance(value, list) and len(value) > 0:
+            return field, len(value)
+
+    return "", 0
+
+
 def validate_record_set(record_set: RecordSet) -> None:
     embeddings = record_set["embeddings"]
     ids = record_set["ids"]
     metadatas = record_set["metadatas"]
 
-    validate_ids(ids)
+    validate_ids(ids) if ids is not None else None
     validate_embeddings(embeddings) if embeddings is not None else None
     validate_metadatas(metadatas) if metadatas is not None else None
 
-    for field, value in record_set.items():
-        if field == "ids" or value is None:
-            continue
+    n_items_field, n_items = get_n_items_from_record_set(record_set)
+    if n_items == 0:
+        return
```

Comment:
how did we get an empty `RecordSet` ? - based on how `get_n_items_from_record_set` works, this must be an empty list - We should probably raise.